src/lottie/hooks/use-lottie.tsx

```diff
@@ -0,0 +1,328 @@
+        const onUnmount = loadAnimation();
+
+        // Clean up on unmount
+        return () => onUnmount?.();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [animationData, loop]);
+
+    // Update the autoplay state
+    React.useEffect(() => {
+        if (!animationInstanceRef.current) {
+            return;
+        }
+
+        animationInstanceRef.current.autoplay = !!autoplay;
+    }, [autoplay]);
+
+    // Update the initial segment state
+    React.useEffect(() => {
+        if (!animationInstanceRef.current) {
+            return;
+        }
+
+        // When null should reset to default animation length
+        if (!initialSegment) {
+            animationInstanceRef.current.resetSegments(true);
+            return;
+        }
+
+        // If it's not a valid segment, do nothing
+        if (!Array.isArray(initialSegment) || !initialSegment.length) {
+            return;
+        }
+
+        // If the current position it's not in the new segment
+        // set the current position to start
+        if (
+            animationInstanceRef.current.currentRawFrame < initialSegment[0] ||
+            animationInstanceRef.current.currentRawFrame > initialSegment[1]
+        ) {
+            animationInstanceRef.current.currentRawFrame = initialSegment[0];
+        }
+
+        // Update the segment
+        animationInstanceRef.current.setSegment(initialSegment[0], initialSegment[1]);
+    }, [initialSegment]);
+
+    /*
+          ======================================
+              EVENTS
+          ======================================
+       */
+
+    /**
+     * Reinitialize listener on change
+     */
+    React.useEffect(() => {
+        const partialListeners: Array<PartialListener> = [
+            {name: 'complete', handler: onComplete},
+            {name: 'loopComplete', handler: onLoopComplete},
+            {name: 'enterFrame', handler: onEnterFrame},
+            {name: 'segmentStart', handler: onSegmentStart},
+            {name: 'config_ready', handler: onConfigReady},
+            {name: 'data_ready', handler: onDataReady},
+            {name: 'data_failed', handler: onDataFailed},
+            {name: 'loaded_images', handler: onLoadedImages},
+            {name: 'DOMLoaded', handler: onDOMLoaded},
+            {name: 'destroy', handler: onDestroy},
+        ];
+
+        const listeners = partialListeners.filter(
+            (listener: PartialListener): listener is Listener => listener.handler !== null
+        );
+
+        if (!listeners.length) {
+            return;
+        }
+
+        const deregisterList = listeners.map((listener: Listener) => {
+            // In Mistica we only use these two events. If we try to add the event listener for all the other
+            // events from the listeners array, some runtime errors appear (and TS complains about wrong types).
+            const misticaAnimationEvents: Array<AnimationEventName> = ['complete', 'loopComplete'];
```

Comment:
I had to add the listeners conditionally only for these two events because I was getting runtime (and TS) errors otherwise.

I've analyzed this and we should be fine by adding only these two events, because we never use the other ones in Mistica